include/fg/chart.h

```diff
@@ -0,0 +1,191 @@
+/*******************************************************
+ * Copyright (c) 2015-2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <fg/defines.h>
+#include <fg/image.h>
+#include <fg/plot.h>
+#include <fg/surface.h>
+#include <fg/histogram.h>
+
+#include <string>
+#include <vector>
+
+namespace internal
```

Comment:
I can move `internal` namespace inside `fg`. If we make `void*` to store class objects, we would have to do `reinterpret_cast` or some sort of casting operation to corresponding class type inside each member function before delegating the call to the corresponding upstream object. Do you have any suggestions for new name to `internal` namespace ?

Regarding `_`: I have been following that convention since the inception. There are three layers in the call delegation starting from the API level to implementation level. It is an easy visual cue as to which layer this object corresponds to and also removes the need to use namespace if both objects are of same name. Using scope resolution results in use of `2+len(namespace)` additional characters instead of just one each time object of this  layer is used. I would like to retain this visual cue, it has proved to be useful.